src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -86,25 +284,44 @@
   ;; See https://poi.apache.org/apidocs/4.1/org/apache/poi/ss/usermodel/DateUtil.html#convertTime-java.lang.String-
   (.setCellValue cell (DateUtil/convertTime (u.date/format "HH:mm:ss" t)))
   (.setCellType cell CellType/NUMERIC)
-  (.setCellStyle cell (cell-style :time)))
+  (.setCellStyle cell (or (cell-style id-or-name) (cell-style :time))))
+
+(defmethod set-cell! OffsetTime
+  [^Cell cell t id-or-name]
+  (set-cell! cell (t/local-time (common/in-result-time-zone t)) id-or-name))
 
-(defmethod spreadsheet/set-cell! OffsetTime
-  [cell t]
-  (spreadsheet/set-cell! cell (t/local-time (common/in-result-time-zone t))))
+(defmethod set-cell! OffsetDateTime
+  [^Cell cell t id-or-name]
+  (set-cell! cell (t/local-date-time (common/in-result-time-zone t)) id-or-name))
 
-(defmethod spreadsheet/set-cell! OffsetDateTime
-  [cell t]
-  (spreadsheet/set-cell! cell (t/local-date-time (common/in-result-time-zone t))))
+(defmethod set-cell! ZonedDateTime
+  [^Cell cell t id-or-name]
+  (set-cell! cell (t/offset-date-time t) id-or-name))
 
-(defmethod spreadsheet/set-cell! ZonedDateTime
-  [cell t]
-  (spreadsheet/set-cell! cell (t/offset-date-time t)))
+(defmethod set-cell! String
+  [^Cell cell value _]
+  (when (= (.getCellType cell) CellType/FORMULA)
```

Comment:
I don't know if this prevents xlsx injection: I will go find out, lol

Reply:
I can get a formula box that gets a shell by noodling around a bit in repl, I am going to try to repro going through the whole thing

Reply:
just curious, how did you test this?

Reply:
if you can get any formula to run upon xlsx open, you can get the excessively "fun" formula to run, including stuff that can exfiltrate data or get shell access

https://labs.bishopfox.com/tech-blog/2018/06/server-side-spreadsheet-injections

docjure seems to take care of it for you luckily enough

Reply:
actually now that i think about it im going to try it on the csvs